### PR TITLE
New module: Grabix check

### PR DESCRIPTION
**Pull request by MartaSantanaSilva**
Initial pull for module grabix/check
Addition of one test (total: 1)

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/grabix/check/main.nf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/grabix:0.1.8--hdcf5f25_9':
+        'biocontainers/grabix:0.1.8--hdcf5f25_9' }"
+
+    input:
+    path(input)
+
+    output:
+    env COMPRESS_BGZIP  , emit: compress_bgzip
+    path "versions.yml"    , emit: versions
```
Comment (nvnieuwk) [inline modules/nf-core/grabix/check/main.nf:14]:
```suggestion
    env COMPRESS_BGZIP  , emit: compress_bgzip
    path "versions.yml" , emit: versions
```
Aligning this makes it easier to be read :)

File: modules/nf-core/grabix/check/main.nf [hunk 2]
```diff
@@ -0,0 +1,30 @@
+process GRABIX_CHECK {
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/grabix:0.1.8--hdcf5f25_9':
+        'biocontainers/grabix:0.1.8--hdcf5f25_9' }"
+
+    input:
+    path(input)
```
Comment (nvnieuwk) [inline modules/nf-core/grabix/check/main.nf:10]:
```suggestion
    tuple val(meta), path(input)
```

nf-core pipelines usually use meta fields alongside their files to keep track of which file it is

File: modules/nf-core/grabix/check/main.nf [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    output:
+    env COMPRESS_BGZIP  , emit: compress_bgzip
```
Comment (nvnieuwk) [inline modules/nf-core/grabix/check/main.nf:13]:
```suggestion
    tuple val(meta), env(COMPRESS_BGZIP)  , emit: compress_bgzip
```
So you output with the same meta as the input to be able to see for what kind of data you just ran the process

File: modules/nf-core/grabix/check/main.nf [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    """
+    COMPRESS_BGZIP=\$(grabix check ${input})
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        grabix: \$(echo \$(grabix | sed -n -E 's/version: (.*)/\1/p'))
+    END_VERSIONS
+    """
+}
```
Comment (nvnieuwk) [inline modules/nf-core/grabix/check/main.nf:30]: Can you add a stub to this module? You can have a look at almost all other modules to have an example for this

File: modules/nf-core/grabix/check/meta.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+tools:
+  - grabix:
+      description: "a wee tool for random access into BGZF files."
+      homepage: "https://github.com/arq5x/grabix"
+      documentation: "https://github.com/arq5x/grabix"
+
+input:
+  - input:
+      type: file
+      description: file to check compression
```
Comment (nvnieuwk) [inline modules/nf-core/grabix/check/meta.yml:17]:
```suggestion
      pattern: "*.gz"
      description: file to check compression 
```

File: modules/nf-core/grabix/check/tests/main.nf.test.snap
Comment (nvnieuwk) [inline modules/nf-core/grabix/check/tests/main.nf.test.snap:1]: Your output channel isn't in the snapshot :grin:

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

File: modules/nf-core/grabix/check/environment.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+name: "grabix_check"
+channels:
+  - conda-forge
+  - bioconda
+  - defaults
+dependencies:
+  - "bioconda::grabix=0.1.8"
```
Comment (maxulysse) [inline modules/nf-core/grabix/check/environment.yml:7]:
```suggestion
  - bioconda::grabix=0.1.8
```

File: modules/nf-core/grabix/check/environment.yml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+name: "grabix_check"
```
Comment (maxulysse) [inline modules/nf-core/grabix/check/environment.yml:1]:
```suggestion
name: grabix_check
```

File: modules/nf-core/grabix/check/main.nf [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/grabix:0.1.8--hdcf5f25_9':
+        'biocontainers/grabix:0.1.8--hdcf5f25_9' }"
+
+    input:
+    tuple val(meta), path(input)
+
+    output:
+    tuple val(meta), stdout   , emit: compress_bgzip
+    path "versions.yml"       , emit: versions
```
Comment (maxulysse) [inline modules/nf-core/grabix/check/main.nf:14]:
```suggestion
    tuple val(meta), stdout, emit: compress_bgzip
    path "versions.yml"    , emit: versions
```

File: modules/nf-core/grabix/check/meta.yml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+      description: file to check compression 
+
+output:
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - compress_bgzip:
+      type: string
+      description: environemental variable with value "yes" or "no"
```
Comment (maxulysse) [inline modules/nf-core/grabix/check/meta.yml:27]: yes/no? why not true/false?

Comment (maxulysse) [reply modules/nf-core/grabix/check/meta.yml:27]:
```suggestion
      description: environmental variable with value "yes" or "no"
```

**Review by maxulysse — APPROVED** [review]

**Review by nvnieuwk — APPROVED** [review]

**Pull request merged**